cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -360,6 +430,14 @@ private predicate simpleInstructionLocalFlowStep(Instruction iFrom, Instruction
   // for now.
   iTo.getAnOperand().(ChiTotalOperand).getDef() = iFrom
   or
+  iTo.getAnOperand().(ChiPartialOperand).getDef() = iFrom.(WriteSideEffectInstruction) and
+  not iTo.isResultConflated()
+  or
+  exists(ChiInstruction chi | iFrom = chi |
+    not chi.isResultConflated() and
+    iTo.(LoadInstruction).getSourceValueOperand().getAnyDef() = chi
+  )
```

Comment:
Please add comments above these two new cases. Recent experience has shown us that cases like these can interact in unexpected ways as we add more, and good comments will ensure we don't accidentally break them if/when we need to adjust them.

Reply:
Good point. I've added this now.

Reply:
That's very helpful. Reading those comments makes me think about how we can improve these rules in follow-up PRs, like allowing for phi nodes between the chi and the load. With these examples in the comments, it should be much harder to accidentally break the desired functionality.